### ISSUE TITLE
Not exactly an issue but... pls help?

### ISSUE THREAD

**Issue by TedKaczynskisCousin** [dragonfireclient/dragonfireclient#45]
I'm quite fond of dragonfire, and I use it in a lot of places it probably shouldn't be used, (lol) but I've recently had the privilege to code mods for a non-anarchy server. (Meaning: hacks = bad, ban)
Although it's quite hard to notice, a large amount of players have either hacked, or have been affected by a hacker. 99% of hackers, use dragonfire, quite obviously because it is very popular.

I have read some of your other posts on github, and you seem to state that dragonfire is built ONLY for anarchy servers, and that you don't condone hacking on other servers.
If so, can you help me out? I'm making an anti-cheat that detects hacks, but I want to make my life a little easier.

I was reading the "Project Exists" Issue #22 page and I noticed something about a "custom version string" that somehow, "is a huge improvement"?
How can I implement this version string in my anticheat?

Thanks

![image](https://user-images.githubusercontent.com/60118488/142001167-5d9dc399-d526-400d-9944-ce3fc5fe4ad5.png)

^^ First mention of "custom version string"

**Comment by fluxionary** [comment]
See my reply @ https://github.com/EliasFleckenstein03/dragonfireclient/issues/22#issuecomment-970550970

As for developing effective anti-cheats, I never got very far with that. The builtin "fast" and "interacted too far" anticheats are effective. The [real suffocation](https://forum.minetest.net/viewtopic.php?t=15304) mod helps a little for noclip, and Gracye's [Catlandia](https://edgy1.net/minetest/cat) server seems to have something even more effective, though a tad overzealous. The "matrix" mode in [Hume2's underch](https://content.minetest.net/packages/Hume2/underch/) is good for keeping players from using the x-ray mode to find ores. I'm not aware of much else that's effective, though I haven't been keeping up much in the last year.

If you're instead looking for an effective way to keep cheaters from returning to the server with minimal impact to well behaved players, I recommend looking at my [verbana](https://github.com/fluxionary/minetest-verbana) mod. 

If you're looking for a good way to catch players in the act of cheating, I recommend [luk3yx's cloaking](https://forum.minetest.net/viewtopic.php?t=20788) mod. 

I hope that helps,
~

**Comment by TedKaczynskisCousin** [comment]
Okay, well I was initially planning to combine different mods together, along with some custom code of my own, specifically instantly detecting dragonfire clients and banning those users with no mercy.
I have already have looked into real suffocation, and I can't access Catlandia's private mods (nor do i really want to.) I'm not entirely sure what this "matrix mode" is you're talking about, so please explain in further detail.
I have already incorporated the cloaking mod, and unfortunately your verbana is only a alternative banning method, not an anticheat per se.

So what now? How is this custom version string a "big improvement" and how will it be useful to me if I cant make it into a mod?
You said something about editing the engine source code, but how will that affect other players joining on the server?
I'm sorry if I'm a bit stupid lol but as far as I am aware, editing the minetest engine is the same as making a custom client, and therefore isnt going to affect anyone else playing, but please correct me if I'm wrong.

TL;DR no it doesn't help, I just have even more questions!! lol

**Comment by fluxionary** [comment]
@LordPhyre let's take this discussion to the minetest forums, I don't think this is really the appropriate place to have it. 

https://forum.minetest.net/viewtopic.php?f=3&t=27465

**Comment by TedKaczynskisCousin** [comment]
good idea. i do hope the mods wont ban us for mentioning various clients such as dragonfire but ok

**Comment by TedKaczynskisCousin** [comment]
(now i need an acc)
